### PR TITLE
Returning Index instead of boolean in knuth_morris_pratt (kmp) function, making it compatible with str.find().

### LINKED ISSUE #9077: Returning index instead of boolean when knuth_morris_pratt matches.
In [knuth morris pratt algorithm](https://github.com/TheAlgorithms/Python/blob/master/strings/knuth_morris_pratt.py), I think it's better to return the "index" of the matched substring in case of success, and `-1` in case of failure, so that it would be consistent with the `str.find()` in Python. `True` and `False` aren't that helpful.

### PR DESCRIPTION
**Pull request by HetarthJain**
### Describe your change:

Returning the index of first occurrence of a match in knuth_morris_pratt algorithm. The return type is converted from boolean to int. The function name parameters are aligned to Python naming conventions. 
Fixes #9077

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a closing keyword : Fixes https://github.com/TheAlgorithms/Python/issues/9077".

### CODE REVIEW

**Comment by rohan472000** [comment]
Kindly tick the checkboxes while raising PR, see contributing guidelines beforehand.

**Comment by HetarthJain** [comment]
> Kindly tick the checkboxes while raising PR, see contributing guidelines beforehand.

I have ticked the appropriate checkboxes.

**Comment by HetarthJain** [comment]
Please review the Pull request

**Comment by rohan472000** [comment]
#9079 is also doing the same, check that first.

**Comment by rohan472000** [comment]
Also keep your PR title descriptive, remove 9077 and add kmp in it, as viewers can understand easily.

**Comment by rohan472000** [comment]
Meanwhile, @HetarthJain you can look into that doctest which is failing in #9079.

**Comment by HetarthJain** [comment]
> Meanwhile, @HetarthJain you can look into that doctest which is failing in #9079.

I have used the doctest in the commit itself, it has passed. Please review it.

**Comment by rohan472000** [comment]
@cclauss kindly have a look at this PR, your doctest for `knuth_morris_pratt` is getting passed here.

File: strings/knuth_morris_pratt.py [hunk 1]
```diff
@@ -61,23 +61,35 @@ def get_failure_array(pattern: str) -> list[int]:
...
-    # Test 5)
+    assert knuth_morris_pratt(text, pattern)
+
+    # Test 5) -> Doctests
+    kmp = "knuth_morris_pratt"
+    assert knuth_morris_pratt(kmp, "kn") == kmp.find("kn")
+    assert knuth_morris_pratt(kmp, "h_m") == kmp.find("h_m")
+    assert knuth_morris_pratt(kmp, "rr") == kmp.find("rr")
+    assert knuth_morris_pratt(kmp, "tt") == kmp.find("tt")
+    assert knuth_morris_pratt(kmp, "not there") == kmp.find("not there")
```
Comment (cclauss) [inline strings/knuth_morris_pratt.py:87]: Remove the duplicates.
```suggestion
```

**Review by cclauss — APPROVED** [review]
I had to add doctests because this PR had none.

https://docs.python.org/3/library/doctest.html

**Comment by HetarthJain** [comment]
Thanks @cclauss  and @rohan472000  for your guidance in my first PR

**Pull request merged**